CLI/turbot.py

```diff
@@ -136,28 +139,67 @@ def parse_filename(file_path):
     prefix = filename.split('.')[0]
     return prefix
 
+def isValidDestination(file_path):
+    print(file_path)
+    if not file_path[-1] == '/':
+        if os.path.isdir(file_path.split('/')[:-2].join('/')):
+            return True
+    else: 
+        if os.isdir(file_path):
+            return True
+    return False
+
+def isPrefixProvided(file_path):
+    if file_path[-1] == '/':
+        return False
+    else: 
+        filename = file_path.split('/')[-1]
+        if not filename.contains('.'):
+            return True
+    return False
+        
 def main():
-    parser = argparse.ArgumentParser(description='Binary Obfuscation Tool (BOT). The flags D, R and O are enabled by default')
+    parser = argparse.ArgumentParser(description='Binary Obfuscation Tool (BOT)')
+    parser.add_argument('-A', action='store_const', const=True, default=False, help='Performs disassembly, obfuscation and reassembly to a given binary')
     parser.add_argument('-D', action='store_const', const=True, default=False, help='Disassembles a specified binary')
     parser.add_argument('-R', action='store_const', const=True, default=False, help='Reassembles a specified assembly file')
     parser.add_argument('-O', action='store_const', const=True, default=False, help='Obfuscates a specified assembly file')
-    parser.add_argument('-p', "--payload", action='append', help='Paths to .s files to be used for obfuscation mode as payloads. This flag can be used multiple times'
+    parser.add_argument('-p', "--payload", action='append', help='Optional: Paths to .s files to be used for obfuscation mode as payloads. This flag can be used multiple times'
                             ' to specify multiple payloads to add. These files must be named for the function to be called inside them. i.e. [func_name].s', metavar='payload', dest='payloads')
-    parser.add_argument('-m', '--mode', action='store', default='lazy', choices=INJECT_MODES, help='The obfuscation mode. Ignored if -O is not present.\n(default: %(default)s)', dest='mode')
-    parser.add_argument('source', metavar='F', type=str, help='The dir where the file is located source/path/file.ext')
-    parser.add_argument('destination', metavar='T', type=str, help='The dir where the output file will be stored output/path/')
+    parser.add_argument('-m', '--mode', action='store', default='lazy', choices=INJECT_MODES, help='Optional: The obfuscation mode. Ignored if -O is not present.\n(default: %(default)s)', dest='mode')
+    parser.add_argument('source', nargs='?', default='', metavar='source', type=str, help='The dir where the file is located source/path/file.ext')
+    parser.add_argument('destination', nargs='?', default=os.getcwd(), metavar='destination', type=str, help='Optional: The dir where the output file will be stored "/output/path/filename-prefix.ext"' 
+                            ' DO NOT include file extension when "-A" flag is set as it is auto generated')
```

Comment:
Not a fan of having a "don't do this" message in our code. Good code should be able to elloquently handle something like this and it makes it look like we're not writing good code.

Few options here:
1. Rephrase to inform users that `-A` will automatically generate filenames and have the program handle it eloquently even if they do provide an extension
2. Use a different way of setting the output filenames that would allow users more control
3. Have the program not generate (or just delete afterwards, I guess) the intermediary files in `-A` mode, allowing the user to name files as intended